urlchecker/client/check.py

```diff
@@ -103,8 +103,13 @@ def main(args, extra):
     # Case 2: We had errors, print them for the user
     if check_results["failed"]:
         print("\n\nDone. The following urls did not pass:")
-        for failed_url in check_results["failed"]:
```

Comment:
I think there must be a better solution here, at least you should sort the list of the results by urls. That way when the same broken link is present in many files, the links are printed in order/ together. A better alternative could be to group all filenames associated with the url and print them all together: `url (fname1, fname2, fname3)`. 

Currently the output is something like this:
```
https://github.com/SuperKogito/URLs-checker/README.md (tests/test_files/sample_test_file.py)
https://none.html (tests/test_files/sample_test_file.md)
https://github.com/SuperKogito/URLs-checker/README.md (tests/test_files/sample_test_file.md)
```
but it should be either 
```
https://github.com/SuperKogito/URLs-checker/README.md (tests/test_files/sample_test_file.py)
https://github.com/SuperKogito/URLs-checker/README.md (tests/test_files/sample_test_file.md)
https://none.html (tests/test_files/sample_test_file.md)
```
or 
```
https://github.com/SuperKogito/URLs-checker/README.md (tests/test_files/sample_test_file.py, tests/test_files/sample_test_file.md)
https://none.html (tests/test_files/sample_test_file.md)
```
@vsoch  which option do you prefer?

Reply:
@SuperKogito I agree that we don't want to redundantly share the same file urls - although I don't love the last UI (what if a URL is in 10 files it will be really hairy looking!) So i think perhaps given that the user wants this verbosity, we should sort based on the filename first, and then just redundantly sort the files. I can also offer a fourth option:

```
https://github.com/SuperKogito/URLs-checker/README.md
    (tests/test_files/sample_test_file.py)
    (tests/test_files/sample_test_file.md)
https://none.html 
    (tests/test_files/sample_test_file.md)
```
which would eliminate some of the redundancy, avoid the awkward printing of stacking multiple URLS into a single line, but still clearly show the URLS and associated files.

Reply:
Alternatively, we could do it the opposite way. That is:
```
tests/test_files/sample_test_file.md
    https://github.com/SuperKogito/URLs-checker/README.md
    https://none.html
```
But I don't care. My horse in this race is for the filenames to show up, but I don't really care how it happens.

Reply:
@mrmundt that's already the format that is printed above! So we've come full circle, lol. The main purpose of this bottom section was to do a summary, hence why we had the short list for failing urls, and then (what I'd normally do) is scroll up to find the files its in.

Reply:
Technically, that is **not** what it does above. Above, it prints out the file name and all URLs that are found in the file. This is a short-list of just files and the URLs that failed in each file.